module_auto_update/models/module.py

```diff
@@ -17,49 +17,65 @@
 class Module(models.Model):
     _inherit = 'ir.module.module'
 
-    checksum_installed = fields.Char(
-        string="Installed Module Checksum",
-    )
     checksum_dir = fields.Char(
-        string="Current Module Checksum",
         compute='_compute_checksum_dir',
     )
-    upgrade_available = fields.Boolean(
-        default=False,
+    checksum_installed = fields.Char(
+        compute='_compute_checksum_installed',
+        store=True,
     )
 
-    @api.multi
+    @api.depends('name')
     def _compute_checksum_dir(self):
+        exclude = self.env["ir.config_parameter"].get_param(
+            "module_auto_update.checksum_excluded_extensions",
+            "pyc,pyo"
+        ).split(",")
+
         for r in self:
-            r.checksum_dir = dirhash(get_module_path(r.name), 'sha1')
+            r.checksum_dir = dirhash(
+                get_module_path(r.name),
+                'sha1',
+                excluded_extensions=exclude
+            )
+
+    @api.depends('state')
+    def _compute_checksum_installed(self):
+        stored_checksums = {}
+        for r in self.read(['checksum_installed']):
+            stored_checksums[r['id']] = r['checksum_installed']
+
+        for r in self:
+            if r.state == 'installed':
+                r.checksum_installed = r.checksum_dir
+            else:
+                r.checksum_installed = stored_checksums[r.id]
+
+    @api.multi
+    def button_uninstall_cancel(self):
+        return self.retain_checksum_installed(
```

Comment:
You don't need these kind of weird hacks if `checksum_installed` is not a computed field.